File: house_robber.py
class Solution:
    # @param num, a list of integer
    # @return an integer
    def rob(self, num):
        dp = [0] * (len(num)+1)
        if len(num) > 0:
            dp[1] = num[0]
        for i in range(2, len(num)+1):
            dp[i] = max(dp[i-1], dp[i-2]+num[i-1])
        return dp[len(num)]

s = Solution()
mainST = [12, 13, 12, 13, 12]
print s.rob(mainST)

File: house_robber.cc
class Solution {
public:
    int rob(vector<int> &num) {
        vector<int> dp(num.size()+1, 0);
        if (num.size() > 0)
            dp[1] = num[0];
        for (int i = 2; i < dp.size(); i++) {
            dp[i] = std::max (dp[i-1], dp[i-2] + num[i-1]);
        }
        return dp[num.size()];
    }
};


